s3-source-connector/src/main/java/io/aiven/kafka/connect/s3/source/S3SourceTask.java

```diff
@@ -174,17 +188,37 @@ private boolean handleException(final RuntimeException exception) throws Interru
         return false;
     }
 
-    private List<SourceRecord> extractSourceRecords(final List<SourceRecord> results) throws InterruptedException {
+    private List<SourceRecord> extractSourceRecords(final List<SourceRecord> results, final int maxRecords) throws InterruptedException {
+        // FIXME : I think the following line is wrong.  We are supposed to return null from Poll if there is no data.
         waitForObjects();
         if (connectorStopped.get()) {
             return results;
         }
-        return RecordProcessor.processRecords(sourceRecordIterator, results, s3SourceConfig, keyConverter,
-                valueConverter, connectorStopped, this.outputWriter, failedObjectKeys);
+        final Map<String, String> conversionConfig = new HashMap<>();
+        int idx = 0;
+        while (idx<maxRecords && aivenS3SourceRecordIterator.hasNext()) {
+            idx++;
+            AivenS3SourceRecord aivenS3SourceRecord = aivenS3SourceRecordIterator.next();
+            final String topic = aivenS3SourceRecord.getToTopic();
+            final Optional<SchemaAndValue> keyData = keyConverter
+                    .map(c -> c.toConnectData(topic, aivenS3SourceRecord.key()));
+
+            outputWriter.configureValueConverter(conversionConfig, s3SourceConfig);
+            valueConverter.configure(conversionConfig, false);
+            try {
+                final SchemaAndValue schemaAndValue = valueConverter.toConnectData(topic, aivenS3SourceRecord.value());
+                results.add(aivenS3SourceRecord.getSourceRecord(topic, keyData, schemaAndValue));
+            } catch (DataException e) {
+                LOGGER.error("Error in reading s3 object stream " + e.getMessage());
```

Comment:
We should probably use LOGGER.error("Error in reading s3 object stream {}", e.getMessage()); or
LOGGER.error("Error in reading s3 object stream ", e);